vortex-file/src/read/buffered.rs

```diff
@@ -0,0 +1,232 @@
+use std::collections::VecDeque;
+use std::io;
+use std::io::ErrorKind;
+use std::pin::Pin;
+use std::sync::{Arc, RwLock};
+use std::task::{Context, Poll, Waker};
+
+use futures::Stream;
+use futures_util::future::BoxFuture;
+use futures_util::{FutureExt, StreamExt};
+use vortex_array::ArrayData;
+use vortex_error::{vortex_err, vortex_panic, VortexExpect, VortexResult};
+use vortex_io::{Dispatch, IoDispatcher, VortexReadAt, VortexReadRanges};
+
+use crate::{LayoutMessageCache, LayoutReader, Message, MessageLocator, MessageRead, RowMask};
+
+const NUM_TO_COALESCE: usize = 8;
+
+pub trait RowMaskReader<V> {
```

Comment:
all of the types in this file should probably either be non-`pub` or at most `pub(crate)` to avoid leaking impl things into public API, now that we're closer to getting some users